axon-server-connector/src/main/java/org/axonframework/axonserver/connector/heartbeat/HeartbeatConnectionCheck.java

```diff
@@ -0,0 +1,89 @@
+package org.axonframework.axonserver.connector.heartbeat;
+
+import org.axonframework.axonserver.connector.AxonServerConnectionManager;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import static io.axoniq.axonserver.grpc.control.PlatformOutboundInstruction.RequestCase.HEARTBEAT;
+
+/**
+ * Implementation of {@link ConnectionSanityCheck} which verifies that heartbeat is properly received.
+ *
+ * @author Sara Pellegrini
+ * @since 4.2
+ */
+public class HeartbeatConnectionCheck implements ConnectionSanityCheck {
+
+    private final ConnectionSanityCheck delegate;
+
+    private final long heartbeatTimeout;
+
+    private final Clock clock;
+
+    private final AtomicReference<Instant> lastReceivedHeartbeat = new AtomicReference<>();
+
+    /**
+     * Constructs an instance of {@link HeartbeatConnectionCheck} using the specified parameters.
+     *
+     * @param connectionManager the connectionManager to AxonServer instance
+     * @param context           the context
+     */
+    public HeartbeatConnectionCheck(AxonServerConnectionManager connectionManager, String context) {
+        this(r -> connectionManager.onOutboundInstruction(context, HEARTBEAT, i -> r.run()),
+             new ActiveGrpcChannelCheck(connectionManager, context));
+    }
+
+    /**
+     * Constructs an instance of {@link HeartbeatConnectionCheck} using a default timeout of 5 seconds and the system clock.
+     * @param registration function which allows to register a callback for the reception of an heartbeat
+     * @param delegate another implementation of {@link ConnectionSanityCheck} that performs others kind of verifications
+     */
+    public HeartbeatConnectionCheck(Consumer<Runnable> registration,
+                                    ConnectionSanityCheck delegate) {
+        this(5_000, registration, delegate, Clock.systemUTC());
```

Comment:
Super nit: I'd replace `5_000` for a named constant, something like `DEFAULT_HEARTBEAT_TIMEOUT_MILLIS`.